web_src/js/features/repo-common.js

```diff
@@ -1,44 +1,42 @@
 import $ from 'jquery';
-import {hideElem, showElem} from '../utils/dom.js';
+import {hideElem, queryElems, showElem} from '../utils/dom.js';
 import {POST} from '../modules/fetch.js';
+import {showErrorToast} from '../modules/toast.js';
 
-async function getArchive($target, url, first) {
-  const dropdownBtn = $target[0].closest('.ui.dropdown.button') ?? $target[0].closest('.ui.dropdown.btn');
-
+async function downloadArchive(target, url, retryCount = 0) {
+  // there are many places using the "archive-link", eg: the dropdown on the repo code page, the release list
+  const targetLoading = target.closest('.ui.dropdown') ?? target;
+  let keepLoading = false;
   try {
-    dropdownBtn.classList.add('is-loading');
+    targetLoading.classList.add('is-loading', 'loading-icon-2px');
```

Comment:
Would add the second class to the element itself in templates. It's not necessary to add/remove with the loading state.

Reply:
There are a lot, and I don't see it's worth to do so.

Reply:
ok